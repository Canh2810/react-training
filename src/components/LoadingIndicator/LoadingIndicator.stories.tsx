import { StoryObj, Meta } from '@storybook/react'

import LoadingIndicator from '.'

export default {
  title: 'Components/LoadingIndicator',
  component: LoadingIndicator,
} as Meta

type Story = StoryObj<typeof LoadingIndicator>

export const Default: Story = {}
