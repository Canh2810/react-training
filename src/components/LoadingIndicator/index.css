.loading__overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(93 93 93 / 50%);
  z-index: 99;
}

.loading__overlay .loader {
  position: absolute;
  left: 50%;
  top: 50%;
}
