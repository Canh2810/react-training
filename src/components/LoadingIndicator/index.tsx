import { CircularProgress } from '../'
import './index.css'

const LoadingIndicator = () => (
  <div className="loading__overlay" data-testid="loading-indicator">
    <CircularProgress />
  </div>
)

export default LoadingIndicator
