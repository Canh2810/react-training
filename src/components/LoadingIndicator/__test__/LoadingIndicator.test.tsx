import { render } from '@testing-library/react'
import LoadingIndicator from '..'

describe('LoadingIndicator Component', () => {
  it('should render correctly', () => {
    const { container } = render(<LoadingIndicator />)
    expect(container).toMatchSnapshot()
  })
})
