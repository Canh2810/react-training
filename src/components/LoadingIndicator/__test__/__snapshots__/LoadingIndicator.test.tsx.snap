// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator Component should render correctly 1`] = `
<div>
  <div
    class="loading__overlay"
    data-testid="loading-indicator"
  >
    <div
      class="loader"
    />
  </div>
</div>
`;
